components/service/glean/src/test/java/mozilla/components/service/glean/storages/DatetimesStorageEngineTest.kt

```diff
@@ -0,0 +1,167 @@
+        )).thenReturn(sharedPreferences)
+
+        storageEngine.applicationContext = context
+        val snapshot = storageEngine.getSnapshot(storeName = "store1", clearStore = true)
+        assertEquals(1, snapshot!!.size)
+        assertEquals("1993-02-23T09:05:23-08:00", snapshot["telemetry.valid_date1"])
+    }
+
+    @Test
+    fun `datetime serializer should correctly serialize datetimes`() {
+        val value = Calendar.getInstance()!!
+        value.set(1993, 1, 23, 9, 5, 23)
+        value.set(Calendar.MILLISECOND, 0)
+        value.setTimeZone(TimeZone.getTimeZone("America/Los_Angeles"))
+
+        run {
+            val storageEngine = DatetimesStorageEngineImplementation()
+            storageEngine.applicationContext = ApplicationProvider.getApplicationContext()
+
+            val metric = DatetimeMetricType(
+                disabled = false,
+                category = "telemetry",
+                lifetime = Lifetime.User,
+                name = "datetime_metric",
+                sendInPings = listOf("store1")
+            )
+
+            // Record the datetime in the store, without providing optional arguments.
+            storageEngine.set(metric, value)
+
+            // Get the snapshot from "store1"
+            val snapshot = storageEngine.getSnapshotAsJSON(storeName = "store1",
+                clearStore = true)
+            // Check that this serializes to the expected JSON format.
+            assertEquals("{\"telemetry.datetime_metric\":\"1993-02-23T09:05:23-08:00\"}",
+                snapshot.toString())
+        }
+
+        // Create a new instance of storage engine to verify serialization to storage rather than
+        // to the cache
+        run {
+            val storageEngine = DatetimesStorageEngineImplementation()
+            storageEngine.applicationContext = ApplicationProvider.getApplicationContext()
+
+            // Get the snapshot from "store1"
+            val snapshot = storageEngine.getSnapshotAsJSON(storeName = "store1",
+                clearStore = true)
+            // Check that this serializes to the expected JSON format.
+            assertEquals("{\"telemetry.datetime_metric\":\"1993-02-23T09:05:23-08:00\"}",
+                snapshot.toString())
+        }
+    }
+
+    @Test
+    fun `setValue() properly sets the value in all stores`() {
+        val storeNames = listOf("store1", "store2")
+
+        val metric = DatetimeMetricType(
+            disabled = false,
+            category = "telemetry",
+            lifetime = Lifetime.Ping,
+            name = "datetime_metric",
+            sendInPings = storeNames
+        )
+
+        // Record the datetime in the stores, without providing optional arguments.
+        DatetimesStorageEngine.set(metric)
+
+        // Check that the data was correctly set in each store.
+        for (storeName in storeNames) {
+            val snapshot = DatetimesStorageEngine.getSnapshot(storeName = storeName, clearStore = false)
```

Comment:
Can we use the test API functions here since you have the `DatetimeMetricType` instance above since you aren't specifically testing `getSnapshot()`?